File: 01_projects/code/java-playwright/cbaps-dawms/robust/src/main/java/com/playwright/cbaps/pages/RoutingApprovalPage.java
package com.playwright.cbaps.pages;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;
import com.playwright.cbaps.library.EnhancedPlaywrightManager;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class RoutingApprovalPage {
    private static final Logger log = LoggerFactory.getLogger(RoutingApprovalPage.class);
    private Page page;
    private EnhancedPlaywrightManager pwm;
    private Locator approverDropdown, submitButton;
    
    public RoutingApprovalPage(Page page, EnhancedPlaywrightManager pwm) {
        this.page = page;
        this.pwm = pwm;
        this.approverDropdown = page.locator("#approver");
        this.submitButton = page.locator("button:has-text('Submit for Approval')");
        pwm.waitUntilElementVisible("#approver");
        log.info("✅ RoutingApprovalPage initialized");
    }
    
    public StatusTrackerPage submitForApproval(String approver) {
        log.info("✍️ Submitting for approval: {}", approver);
        pwm.selectDropdown(approverDropdown, approver);
        pwm.clickElement(submitButton);
        return new StatusTrackerPage(page, pwm);
    }
}

File: 01_projects/code/python-playwright/cbaps-dawms/robust/cbaps/pages/routing_approval_page.py
from playwright.sync_api import Page
from loguru import logger

class RoutingApprovalPage:
    def __init__(self, page: Page, pw):
        self.page = page
        self.pw = pw
        self.approver_dropdown = page.locator("#approver")
        self.submit_btn = page.locator("button:has-text('Submit for Approval')")
        self.pw.wait_for_selector("#approver")
    
    def submit_for_approval(self, approver: str):
        from .status_tracker_page import StatusTrackerPage
        logger.info(f"✍️ Submitting: {approver}")
        self.pw.select_dropdown(self.approver_dropdown, approver)
        self.pw.click_element(self.submit_btn)
        return StatusTrackerPage(self.page, self.pw)
